### PR TITLE
sd: omit zero-initialization for local variable and add config SD_CMD_RETRIES

### PR DESCRIPTION
**Pull request by ndrs-pst**
This series of commits is an effort to remove zero-initialization, specifically targeting the `sdhc_command` and `sdhc_data` structures.
Suggestions and insights from maintainers are always welcome and appreciated. 🚀

### CODE REVIEW

**Comment by decsny** [comment]
can I ask why?

**Comment by ndrs-pst** [comment]
@decsny, From my point of view, the decision to omit zero-initialization from the `sdhc_command` and `sdhc_data` structures is driven by a desire to enhance Zephyr's compactness and efficiency.

I acknowledge that the definitions of 'compact' and 'efficient' can vary from person to person.
However, in the context of Zephyr, I believe this change contributes to reducing unnecessary operations, thereby streamlining the codebase.

**Comment by decsny** [comment]
> @decsny, From my point of view, the decision to omit zero-initialization from the `sdhc_command` and `sdhc_data` structures is driven by a desire to enhance Zephyr's compactness and efficiency.
> 
> I acknowledge that the definitions of 'compact' and 'efficient' can vary from person to person.
> However, in the context of Zephyr, I believe this change contributes to reducing unnecessary operations, thereby streamlining the codebase.

Can you just put in the commit message the reason you're doing it, like if you think it reduces instruction count or any other reason

**Comment by ndrs-pst** [comment]
> Can you just put in the commit message the reason you're doing it, like if you think it reduces instruction count or any other reason

@decsny, I have revised the commit message.
Could you please review the updated message at at https://github.com/zephyrproject-rtos/zephyr/pull/68017/commits/f7406d65fd33b428ab18b8a40d05e5728a3db1e0 ? Thank you.

File: subsys/sd/sd_ops.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 /*
  * Switches voltage of SD card to 1.8V, as described by
- * "Signal volatage switch procedure" in section 3.6.1 of SD specification.
+ * "Signal voltage switch procedure" in section 3.6.1 of SD specification.
```
Comment (danieldegrasse) [inline subsys/sd/sd_ops.h:13]: Thanks for catching this- while you are at it, it looks like the section is now named "Signal Voltage Switch Sequence". We should also drop the section number, that may change with revisions of the specification

**Review by danieldegrasse — COMMENTED** [review]
This does save a decent amount of flash space- when building `tests/subsys/sd/sdmmc` on the RT1060, utilized flash space goes from 71168 Bytes on sha 04de43b1a119665 to 71016 Bytes on sha c716b805 (head of this PR)

File: subsys/sd/sd_ops.h (hunk 1 above)
Comment (danieldegrasse) [reply subsys/sd/sd_ops.h:13]: I was indeed looking at the wrong spec...apologies, shouldn't be commenting on GitHub so early in the morning 😆. Could you change "SD specification" to "SD host controller specification" to protect me from myself? I do want to refer the section 3.6.1 in the SD host controller simplified specification

Comment (ndrs-pst) [reply subsys/sd/sd_ops.h]: Already push another commit

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Pull request merged**